client/components/App.js

```diff
@@ -1,21 +1,51 @@
 import React, { Component } from 'react';
+import {connect} from 'react-redux';
+import {browserHistory} from 'react-router';
 
 // components
 import Header from './common/Header';
 import Footer from './common/Footer';
 
+// stylesheet
 import '../assets/style.scss';
 
 class App extends Component {
+
+  componentDidMount(){
+    if (this.props.currentUrl === "/" && this.props.isAuthenticated) {
+      browserHistory.replace("/home");
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { redirectUrl } = this.props;
+    const isLoggingOut = prevProps.isAuthenticated && !this.props.isAuthenticated;
+    const isLoggingIn = !prevProps.isAuthenticated && this.props.isAuthenticated;
+
+    if (isLoggingIn) {
+      redirectUrl ? browserHistory.replace(redirectUrl) : browserHistory.replace("/home");
+    } else if (isLoggingOut) {
+      browserHistory.replace("/");
+    }
+  }
+
   render() {
     return (
       <div className="wrapper">
-        {this.props.location.pathname === '/' ? null : <Header />}
+        {this.props.location.pathname !== '/' && <Header />}
```

Comment:
What does this do? Not sure I understand @andela-iukwuoma

Reply:
It simply checks if the first statement is true then it execute the second statement. So the second statement will only execute if the first statement is true. I felt its more concise, that why I changed it.

Reply:
Haha! Gotcha @andela-iukwuoma 
Smart!